src/Tasks/Microsoft.NET.Build.Tasks/NuGetPackageResolver.cs

```diff
@@ -31,6 +31,19 @@ public string GetPackageDirectory(string packageId, NuGetVersion version)
             return _packagePathResolver.GetPackageDirectory(packageId, version);
         }
 
+        public string GetPackageDirectory(string packageId, NuGetVersion version, out string packageRoot)
+        {
+            string packageDir = null;
+            packageRoot = null;
+            var pkginfo = _packagePathResolver.GetPackageInfo(packageId,version);
+            if (pkginfo != null)
+            {
+                packageDir  = pkginfo.PathResolver.GetInstallPath(packageId, version);
+                packageRoot = pkginfo.PathResolver.GetVersionListPath("");  //TODO Check with nuget  if this is the correct way to get the package root
```

Comment:
This is a hack for now, i filed an issue with nuget client and got them to get a change- https://github.com/NuGet/NuGet.Client/pull/1096, that we will use as
FallbackPackagePathInfo.PathResolver.RootPath to get the packageRoot.